src/gui/properties/speedwidget.cpp

```diff
@@ -116,52 +116,38 @@ SpeedWidget::SpeedWidget(PropertiesWidget *parent)
 
     loadSettings();
 
-    m_isUpdating = true;
-    m_updateFuture = QtConcurrent::run(this, &SpeedWidget::update);
+    QTimer *localUpdateTimer = new QTimer(this);
```

Comment:
I don't understand why we really need this timer. Why can't we update it synchronously with main update event in BitTorrent::Session?

Reply:
I think it was made so because main update event has arbitrary user tunable period. So if we updates on this event, math will become more complicated (especially averaging part (it already have errors in it)). We can not use compile/design time calculated values for dividers and so on. Also this period can be changed (theoretically) in the middle of program run and we have to coupe with this. 
According to it I think that dedicated timing construction for it is much simpler in realization.

Reply:
>I think it was made so because main update event has arbitrary user tunable period. So if we updates on this event, math will become more complicated

I don't think it becomes too complicated...
>it already have errors in it

Please detail!

Reply:
At file  qBittorrent/src/gui/properties/speedplotview.cpp there is function **SpeedPlotView::pushPoint**.
At this function there is averaging code kind of that at line 95
```
m_data30Min.back().y[id] = (m_data30Min.back().y[id] * m_counter30Min + point.y[id]) / (m_counter30Min + 1);
```
This code is not transparent and at first look I thought that it calculates biased average. It several 5 min for me to evaluate this code and insure that there is *no* error. 
I'm sorry men, there is no error, just obscure code.

Reply:
>This code is not transparent and at first look I thought that it calculates biased average.

I would extract it into function.

Reply:
@glassez push commit with averaging code rewrite. Don't know, may be separate PR would be better for it? Anyway, take a look, please.

Reply:
I don't see you include QTimer header.

Reply:
Fixed.